pyvrp/Population.py

```diff
@@ -15,38 +13,34 @@ class Population:
 
     Parameters
     ----------
-    data
-        Data object describing the problem to be solved.
-    penalty_manager
-        Penalty manager to use.
     rng
         Random number generator.
     diversity_op
         Operator to use to determine pairwise diversity between solutions. Have
         a look at :mod:`pyvrp.diversity` for available operators.
+    initial_solutions
+        List of individuals used to initialize the population.
     params, optional
         Population parameters. If not provided, a default will be used.
     """
 
     def __init__(
         self,
-        data: ProblemData,
-        penalty_manager: PenaltyManager,
         rng: XorShift128,
         diversity_op: Callable[[Individual, Individual], float],
+        initial_solutions: List[Individual],
         params: PopulationParams = PopulationParams(),
     ):
-        self._data = data
-        self._pm = penalty_manager
         self._rng = rng
```

Comment:
We can also remove the RNG dependency and pass it as argument to `self.select`/`self.binary_tournament`. I don't know if that's desirable.

Reply:
I think it's better because there's less internal state on `Population` then.